src/main/java/com/aws/greengrass/mqttclient/v5/Unsubscribe.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.mqttclient.v5;
+
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+
+import java.util.function.Consumer;
+
+@Builder
+@Value
+public class Unsubscribe {
+    @NonNull String topic;
+    @NonNull Consumer<Publish> callback;
```

Comment:
Should the callback accept a `Publish`? Or do we need an unsuback?

Reply:
this callback isn't being called, rather it is being used to remove the correct callback from the client's internal mapping. I don't think we really care about the unsuback as a user of the Nucleus mqtt client. The nucleus can and should abstract any failures of that.

Reply:
Updated field name and commented.